langchain-core/src/prompts/image.ts

```diff
@@ -0,0 +1,186 @@
+   * The format of the prompt template. Options are 'f-string'
+   *
+   * @defaultValue 'f-string'
+   */
+  templateFormat?: TemplateFormat;
+
+  /**
+   * Whether or not to try validating the template on initialization
+   *
+   * @defaultValue `true`
+   */
+  validateTemplate?: boolean;
+}
+
+/**
+ * An image prompt template for a multimodal model.
+ */
+export class ImagePromptTemplate<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  RunInput extends InputValues = any,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  PartialVariableName extends string = any
+> extends BasePromptTemplate<RunInput, ImagePromptValue, PartialVariableName> {
+  static lc_name() {
+    return "ImagePromptTemplate";
+  }
+
+  template: Record<string, unknown>;
+
+  templateFormat: TemplateFormat = "f-string";
+
+  validateTemplate = true;
+
+  constructor(input: ImagePromptTemplateInput<RunInput, PartialVariableName>) {
+    super(input);
+    this.template = input.template;
+    this.templateFormat = input.templateFormat ?? this.templateFormat;
+    this.validateTemplate = input.validateTemplate ?? this.validateTemplate;
+
+    if (this.validateTemplate) {
+      let totalInputVariables: string[] = this.inputVariables;
+      if (this.partialVariables) {
+        totalInputVariables = totalInputVariables.concat(
+          Object.keys(this.partialVariables)
+        );
+      }
+      /** @TODO Fix this */
+      // checkValidTemplate(
+      //   this.template,
+      //   this.templateFormat,
+      //   totalInputVariables
+      // );
+    }
+  }
+
+  _getPromptType(): "prompt" {
+    return "prompt";
+  }
+
+  /**
+   * Partially applies values to the prompt template.
+   * @param values The values to be partially applied to the prompt template.
+   * @returns A new instance of ImagePromptTemplate with the partially applied values.
+   */
+  async partial<NewPartialVariableName extends string>(
+    values: PartialValues<NewPartialVariableName>
+  ) {
+    const newInputVariables = this.inputVariables.filter(
+      (iv) => !(iv in values)
+    ) as Exclude<Extract<keyof RunInput, string>, NewPartialVariableName>[];
+    const newPartialVariables = {
+      ...(this.partialVariables ?? {}),
+      ...values,
+    } as PartialValues<PartialVariableName | NewPartialVariableName>;
+    const promptDict = {
+      ...this,
+      inputVariables: newInputVariables,
+      partialVariables: newPartialVariables,
+    };
+    return new ImagePromptTemplate<
+      InputValues<
+        Exclude<Extract<keyof RunInput, string>, NewPartialVariableName>
+      >
+    >(promptDict);
+  }
+
+  /**
+   * Formats the prompt template with the provided values.
+   * @param values The values to be used to format the prompt template.
+   * @returns A promise that resolves to a string which is the formatted prompt.
+   */
+  async format<FormatOutput = ImageURL>(
+    values: TypedPromptInputValues<RunInput>
+  ): Promise<FormatOutput> {
```

Comment:
This can't be right? Or is `ImagePromptTemplate` just the image content part?

Reply:
`ImagePromptTemplate` I don't believe is ever intended to be used by an end user, and format returning the `ImageURL` type is accurate since that's the type which matches the OpenAI sdk:

```typescript
export interface ImageURL {
  /** Specifies the detail level of the image. */
  detail?: "auto" | "low" | "high";

  /** Either a URL of the image or the base64 encoded image data. */
  url: string;
}
```